control-plane/api-gateway/common/resources.go

```diff
@@ -364,6 +365,35 @@ func (s *ResourceMap) ReferenceCountHTTPRoute(route gwv1beta1.HTTPRoute) {
 	s.httpRouteGateways[consulKey] = set
 }
 
+func localObjectReferenceToObjectReference(filterRef gwv1beta1.LocalObjectReference, namespace string) corev1.ObjectReference {
+	return corev1.ObjectReference{
+		Kind:      string(filterRef.Kind),
+		Name:      string(filterRef.Name),
+		Namespace: namespace,
+	}
+}
+
+func (s *ResourceMap) AddExternalFilter(filterRef gwv1beta1.LocalObjectReference, filter client.Object) {
```

Comment:
Since you have the full object in this case, you could just construct the key from it a la:

```
 corev1.ObjectReference{
    Kind:      filter.GetKind(),
    Name:      filter.GetName(),
    Namespace: filter.GetNamespace(),
}
```

And drop the need to pass the `filterRef` in here at all. I'll leave it up to you as to whether or not you want to change it though, as it's at least somewhat shaped like the "get" methods.